lib/kubernetes-deploy/duration_parser.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module KubernetesDeploy
+  ##
+  # This class is a less strict extension of ActiveSupport::Duration::ISO8601Parser.
+  # In addition to full ISO8601 durations, it can parse unprefixed ISO8601 time components (e.g. '1H').
+  # It is also case-insensitive.
+  # For example, this class considers the values "1H", "1h" and "PT1H" to be valid and equivalent.
+
+  class DurationParser
+    class ParsingError < ArgumentError; end
+
+    def initialize(value)
+      @iso8601_str = value.to_s.strip.upcase
```

Comment:
I believe if you add:
```
@iso8601_str = "PT" + @iso8601_str unless @iso8601_str.start_with?("P")
```

You get all of the features you want from just normal `ActiveSupport::Duration.parse(@iso8601_str)` without needing to reach into the parser or write `calculate_total_seconds`.

The only issue is that the error message on a failed parse becomes `Invalid ISO 8601 duration: "PTFOO"`, which is not really what the user provided.

You could maybe solve this with a sketchy Find&Replace on the returned message... maybe

Or you could do a more complicated thing where you do:

      begin
        ActiveSupport::Duration.parse(@iso8601_str)
      rescue ActiveSupport::Duration::ISO8601Parser::ParsingError => e
        begin
          ActiveSupport::Duration.parse("PT" + @iso8601_str)
        rescue ActiveSupport::Duration::ISO8601Parser::ParsingError
          raise ParsingError, e.message
        end
      end

Where you do the blind parse, and if it fails you always try jamming `PT` on the front.
If that succeeds, we return that one, and if it also fails, we return the first failure and discard the second (We raise the `e.message`, which is the first exception, in the second rescue block and don't even capture the second exception)

Reply:
I like the retry-and-append solution! The only thing that bugs me about it is that we'll be retying in what I'm expecting to be the average case. I can't imagine that's meaningfully expensive, but for the purposes of reflecting intent, I'd slightly prefer to have the first try use the `PT +` version.